membership.lua

```diff
@@ -448,11 +469,17 @@ local function protocol_step()
         end
     end
 
-    if sent_indirect > 0 and wait_ack(uri, loop_now, opts.PROTOCOL_PERIOD_SECONDS * 1.0e6) then
-        local member = members.get(uri)
-        members.set(uri, member.status, member.incarnation)
-        return
-    elseif members.get(uri).status == opts.ALIVE then
+    if sent_indirect > 0 then
+        local ok, ack_data = wait_ack(uri, loop_now, opts.PROTOCOL_PERIOD_SECONDS * 1.0e6)
+        if ok then
+            local member = members.get(uri)
+            -- calculate time difference between local time and member time
+            local delta = _get_clock_delta(ack_data)
+            members.set(uri, member.status, member.incarnation, { clock_delta = delta })
+            return
+        end
+    end
+    if members.get(uri).status == opts.ALIVE then
```

Comment:
`elseif` was important. Imagine you received ack after indirect ping. Don't mark it suspect.

Reply:
If we receive indirect ACK, we return from protocol_step 2 lines above
But ok, I'll change this logic to get closer to original code

Reply:
You're right